plans/splitbrain/main.go

```diff
@@ -0,0 +1,172 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"net/http"
+	"time"
+
+	"github.com/testground/sdk-go/network"
+	"github.com/testground/sdk-go/runtime"
+	"github.com/testground/sdk-go/sync"
+)
+
+type region int
+
+const (
+	regionA = iota
+	regionB
+	regionC
+)
+
+func (r region) String() string {
+	return [...]string{"region_A", "region_B", "region_C"}[r]
+}
+
+type node struct {
+	Region region
+	IP     *net.IP
+}
+
+func main() {
+	testcases := map[string]runtime.TestCaseFn{
+		"drop":   routeFilter(network.Drop),
+		"reject": routeFilter(network.Reject),
+		"accept": routeFilter(network.Accept),
+	}
+	runtime.InvokeMap(testcases)
+}
+
+func expectErrors(runenv *runtime.RunEnv, a *node, b *node) bool {
+	if runenv.TestCase == "accept" || a.Region == regionC || b.Region == regionC {
+		return false
+	}
+	if (a.Region == regionA && b.Region == regionB) || (a.Region == regionB && b.Region == regionA) {
+		return true
+	}
+	return false
+}
+
+func routeFilter(action network.FilterAction) runtime.TestCaseFn {
+
+	return func(runenv *runtime.RunEnv) error {
+
+		ctx, cancel := context.WithTimeout(context.Background(), 300*time.Second)
+		defer cancel()
+
+		client := sync.MustBoundClient(ctx, runenv)
+
+		if !runenv.TestSidecar {
+			return fmt.Errorf("this plan must be run with sidecar enabled")
+		}
+
+		netclient := network.NewClient(client, runenv)
+		netclient.MustWaitNetworkInitialized(ctx)
+
+		// Each node starts an HTTP server to test for connectivity
+		runenv.RecordMessage("Starting http server")
+		http.HandleFunc("/", func(w http.ResponseWriter, req *http.Request) {
+			runenv.RecordMessage("received http request from %s", req.RemoteAddr)
+			fmt.Fprintln(w, "hello.")
+		})
+		go func() {
+			err := http.ListenAndServe(":8765", nil)
+			runenv.RecordFailure(err)
```

Comment:
We will always reach this point, so I am not sure why we always want to `RecordFailure` ?